react/src/components/UserPrefModal.tsx

```diff
@@ -0,0 +1,300 @@
+import React, { useEffect } from 'react';
+import { Modal, Input, Form, Select, Divider, message, Switch } from 'antd';
+import { useTranslation } from "react-i18next";
+import { useWebComponentInfo } from './DefaultProviders';
+import { passwordPattern } from './ResetPasswordRequired';
+import { useSuspendedBackendaiClient } from "../hooks";
+import { baiSignedRequestWithPromise } from '../helper';
+import { useTanMutation } from '../hooks/reactQueryAlias';
+
+const UserPrefModal : React.FC = () => {
+  const { t } = useTranslation();
+
+  const [form] = Form.useForm();
+
+  const [messageApi, contextHolder] = message.useMessage();
+  const _showSuccessMessage = (successMessage: string) => {
+    messageApi.open({
+      type: "success",
+      content: successMessage
+    });
+  };
+  const _showErrorMessage = (errorMessage: string) => {
+    messageApi.open({
+      type: "error",
+      content: errorMessage
+    })
+  }
+
+  const { value, dispatchEvent } = useWebComponentInfo();
+  let parsedValue: {
+    isOpen: boolean;
+    full_name: string;
+    userId: string;
+    totpSupported: boolean;
+    totpActivated: boolean;
+    loggedAccount: {
+      access_key: string;
+    };
+    keyPairInfo: {
+      keypairs: [{
+        access_key: string;
+        secret_key: string;
+      }];
+    }
+  };
```

Comment:
It's better to query user info in this component rather than getting it from the props. Please replace these lines with querying using graphQL or `useSuspendedBackendaiClient`.